### DISCUSSION TITLE
execute command in operator

### DISCUSSION THREAD

**Discussion by tihamc** [sqlalchemy/sqlalchemy#10303] [Usage Questions]
Hi,

I have a LIST of string and i wanna bind this list to an IN operator param im my SQL for execute in a session.execute(sql, param) function. But its not working. I can´t find the sintax. I need some help.

Ex:

Query file
select riblocos.nome nm_bloco,
       ricadpac.nome nm_paciente
from ricadint
inner join riblocos on riblocos.bloco = ricadint.bloco
inner join ricadpac on ricadpac.pront = ricadint.pront
where ricadint.alta = '30.12.1899'
and riblocos.nome in (:_bloco)

Python
file = blpainel.open_resource('sgh_sql/paciente_internado.sql', 'r')
sql = text(file.read())
param = {'_bloco': ['B1','B2']}
pi = db.session.execute(
        sql,
        params=param,
        bind_arguments={'bind': db.get_engine('sgh')}
    ).fetchall()

**Comment by zzzeek** [comment]
you can't bind a list to a single parameter like that with textual SQL.  the IN expression must be expanded out in order to accommodate each value in the the list that you have, such as:

```py

stmt = text("SELECT * FROM table WHERE col IN (:param1, :param2)")
params = {"param1": "B1", "param2": "B2"}

```

if you make use of [the SQL expression language](https://docs.sqlalchemy.org/en/20/tutorial/data_select.html#the-select-sql-expression-construct) then this expansion is performed automatically:

```py

stmt = select(table).where(table.c.col.in_(["B1", "B2"])
````

**Reply by CaselIT** [reply]
to expand on what mike said, if you are using psycopg2 it may be possible to what you are trying (that's a oddity of that dbapi; in general it's not supported)
